### PR TITLE
feat(date-picker): add close-on-select prop.

### PR DESCRIPTION
**Pull request by kungege**
close [#541](https://github.com/TuSimple/naive-ui/issues/541)

### CODE REVIEW

File: CHANGELOG.en-US.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 - `n-checkbox-group` add `min` and `max` prop.
 - `n-mention` add `empty` slot.
 - `n-space` `justify` prop supports `center`, `space-around` and `space-between`.
+- `n-date-picker` add `close-on-selected` prop, closes [#541](https://github.com/TuSimple/naive-ui/issues/541).
```
Comment (doom-9) [inline CHANGELOG.en-US.md:12]: 😄

**Comment by kungege** [comment]
waiting for [#578](https://github.com/TuSimple/naive-ui/pull/578).

File: src/date-picker/src/DatePicker.tsx [hunk 2]
```diff
@@ -107,6 +107,10 @@ const datePickerProps = {
     default: undefined
   },
   ranges: Object as PropType<Record<string, [number, number]>>,
+  closeOnSelected: {
+    type: Boolean,
+    default: false
+  },
```
Comment (07akioni) [inline src/date-picker/src/DatePicker.tsx:113]:
```suggestion
  closeOnSelected: Boolean,
```

**Review by 07akioni — COMMENTED** [review]
3 Things:
1. `close-on-selected` => `close-on-select`
2. Only add the prop to daterange typed, adding it to `datetimerage` seems to have no sense (user need to check time).
3. Check if it has conflict with `update-value-on-close`, only 1 `update:value` should be submitted.

**Pull request merged**